ccm_web/client/src/App.tsx

```diff
@@ -32,8 +35,23 @@ function App (props: AppProps): JSX.Element {
   const classes = useStyles()
   const features = allFeatures.map(f => f.features).flat()
 
-  const [globals, isAuthenticated, loading, error] = useGlobals(props.ltiKey)
-  if (isAuthenticated === undefined || loading) return <div className='App'><p>Loading...</p></div>
+  const [globals, isAuthenticated, isGlobalsLoading, error] = useGlobals(props.ltiKey)
+
+  const [course, setCourse] = useState<undefined|CanvasCourseBase>(undefined)
+  const [doLoadCourse, isCourseLoading, getCourseError] = usePromise<CanvasCourseBase|undefined, typeof getCourse>(
+    async (ltiKey: string|undefined, courseId: number): Promise<CanvasCourseBase> => {
+      return await getCourse(ltiKey, courseId)
+    },
+    (value: CanvasCourseBase|undefined) => setCourse(value)
+  )
+
+  useEffect(() => {
+    if (globals !== undefined) {
+      void doLoadCourse(props.ltiKey, globals.course.id)
+    }
+  }, [isGlobalsLoading])
```

Comment:
I think we'll get the same effect but trigger this hook less often if we subscribe to changes to `globals` instead of `isGlobalsLoading`. With the latter we get three triggers (on mount, going to `true`, going to `false`), where as with the former, we just get two (on mount, going from `undefined` to `{...}`). Minor detail, but it occurred to me.

Reply:
I'm interested in this, I don't know the lifecycle well.  When I put a debugging statement in it seemed like this only got called once, and when I switched it to filter on changes to globals it got called  twice.  Was this a theoretical thing or did you observe something?

Reply:
I tested this, but could have missed something. Try putting a log statement inside `useEffect` but before the `globals !== undefined` check, and count how many times it's logged when you launch. I saw the 2 versus 3 difference, I thought.

Reply:
You're right, it was 3 vs 2, when it was 3 chrome grouped them up and put a 3 next to it that I missed.  I can switch it to use globals.